distributed.go

```diff
@@ -0,0 +1,174 @@
+package ffi
+
+import (
+	"github.com/filecoin-project/filecoin-ffi/generated"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/specs-actors/actors/runtime/proof"
+	"github.com/pkg/errors"
+)
+
+type FallbackChallanges struct {
+	Sectors    []abi.SectorNumber
+	Challanges map[abi.SectorNumber][]uint64
+}
+
+type VanillaProof []byte
+
+// GenerateWinningPoStSectorChallenge
+func GenerateWinningPoStFallbackSectorChallenges(
+	proofType abi.RegisteredPoStProof,
+	minerID abi.ActorID,
+	randomness abi.PoStRandomness,
+	sectorIds []abi.SectorNumber,
+) (*FallbackChallanges, error) {
+	proverID, err := toProverID(minerID)
+	if err != nil {
+		return nil, err
+	}
+
+	pp, err := toFilRegisteredPoStProof(proofType)
+	if err != nil {
+		return nil, err
+	}
+
+	secIds := make([]uint64, 0, len(sectorIds))
+	for i, sid := range sectorIds {
+		secIds[i] = uint64(sid)
+	}
+
+	resp := generated.FilGenerateFallbackSectorChallenges(
+		pp, to32ByteArray(randomness), secIds, uint(len(secIds)),
+		proverID,
+	)
+	resp.Deref()
+	resp.IdsPtr = resp.IdsPtr[:resp.IdsLen]
+	resp.ChallengesPtr = resp.ChallengesPtr[:resp.ChallengesLen]
+
+	defer generated.FilDestroyGenerateFallbackSectorChallengesResponse(resp)
+
+	if resp.StatusCode != generated.FCPResponseStatusFCPNoError {
+		return nil, errors.New(generated.RawString(resp.ErrorMsg).Copy())
+	}
+
+	// copy from C memory space to Go
+
+	var out FallbackChallanges
+	out.Sectors = make([]abi.SectorNumber, resp.IdsLen)
+	stride := int(resp.ChallengesStride)
+	for idx := range resp.IdsPtr {
+		secNum := abi.SectorNumber(resp.IdsPtr[idx])
+		out.Sectors[idx] = secNum
+		out.Challanges[secNum] = append([]uint64{}, resp.ChallengesPtr[idx*stride:(idx+1)*stride]...)
```

Comment:
Hate to nitpick on ffi code, but `out.Challanges` is a typo (spelling).

Reply:
It is just me typing, I need good spell correct for code ;p